### PR TITLE
add owner_id field [RHCLOUD-8583]

### PR DESCRIPTION
**Pull request by ryandillinfelton**
Mirror of changes needed for https://github.com/RedHatInsights/insights-host-inventory/pull/747/files.
Adds a new field to the sytem profile `owner_id` which is validated as a UUID

### CODE REVIEW

**Review by kruai — APPROVED** [review]

File: schemas/system_profile/v1.yaml [hunk 1]
```diff
@@ -150,6 +150,9 @@ $defs:
     description: Representation of the system profile fields
     type: object
     properties:
+      owner_id:
+        type: string
+        pattern: '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}'
```
Comment (tahmidefaz) [inline schemas/system_profile/v1.yaml:155]: Can we have an example of how an `owner_id` might look like?

**Review by tahmidefaz — CHANGES_REQUESTED** [review]

File: schemas/system_profile/v1.yaml (hunk 1 above)
Comment (kruai) [reply schemas/system_profile/v1.yaml:155]: Good point, Tahmid... That would make a good reference, and help out the regex-impaired (like me)

Comment (ryandillinfelton) [reply schemas/system_profile/v1.yaml]: Added an example and a description

**Comment by tahmidefaz** [comment]
@ryandillinfelton for some reason the pr check isn't working for this repo. I had some doubts, so I decided to run the lint check locally on my laptop, and I got:
`153:16  error  property-string-max-length  #/$defs/SystemProfile/properties/owner_id does not define max length`

**Review by thearifismail — APPROVED** [review]

**Comment by ryandillinfelton** [comment]
Looks like the test is still busted. Closing and reopening [here](https://github.com/RedHatInsights/inventory-schemas/pull/34) to switch the check.

**Pull request closed without merging**